test/engine/SpatialJoinTest.cpp

```diff
@@ -304,193 +236,346 @@ void testGetResultWidthOrVariableToColumnMap(bool leftSideBigChild,
+                            bool addDist, bool withObj, bool withName,
+                            bool withGeo) -> std::string {
+      size_t currentColumn = 0;
+      std::ostringstream expected;
+
+      auto addV = [&](std::vector<std::string> vec, bool right = false) {
+        std::ostringstream expected_inner;
+        for (auto v : vec) {
+          if (right &&
+              ((v == "?obj" && !withObj) || (v == "?name" && !withName) ||
+               (v == "?geo" && !withGeo))) {
+            continue;
+          }
+          expected_inner << "(" << v << (right ? "2" : "1") << " => "
+                         << currentColumn << ")";
+          currentColumn++;
+        }
+        return expected_inner.str();
+      };
+
+      // Left
+      expected << (leftSideBigChild ? addV(bigChild) : addV(smallChild))
+               << addV({"?point"})
+               // Right
+               << (rightSideBigChild ? addV(bigChild, true)
+                                     : addV(smallChild, true))
+               << addV({"?point"}, "2");
+      if (addDist) {
+        expected << "(" << distVar.name() << " => " << currentColumn << ")";
+      }
+      return expected.str();
+    };
+
+    for (bool withObj : bools) {
+      for (bool withGeo : bools) {
+        for (bool withName : bools) {
+          for (bool addDist : bools) {
+            std::vector<Variable> payloadVars;
+            if (withObj) {
+              payloadVars.push_back(Variable{"?obj2"});
+            }
+            if (withGeo) {
+              payloadVars.push_back(Variable{"?geo2"});
+            }
+            if (withName && rightSideBigChild) {
+              payloadVars.push_back(Variable{"?name2"});
+            }
+
+            // Test the regular and valid payloadVars
+            auto exp = makeExpected(leftSideBigChild, rightSideBigChild,
+                                    addDist, withObj, withName, withGeo);
+            computeAndCompareVarToColMaps(addDist, payloadVars, exp);
+
+            // Also test the PayloadAllVariables version
+            if (withGeo && withObj && (withName || !rightSideBigChild)) {
+              computeAndCompareVarToColMaps(addDist, PayloadAllVariables{},
+                                            exp);
+            }
+
+            // Test multiple occurrences of variables
+            if (withObj) {
+              payloadVars.push_back(Variable{"?obj2"});
+              // Variable ?obj2 is now contained twice
+              computeAndCompareVarToColMaps(addDist, payloadVars, exp);
+              payloadVars.pop_back();
+            }
+
+            // Test contained right variable in payloadVars
+            payloadVars.push_back(Variable{"?point2"});
+            computeAndCompareVarToColMaps(addDist, payloadVars, exp);
+            payloadVars.pop_back();
+
+            // Test throw if left join variable contained
+            payloadVars.push_back(Variable{"?point1"});
+            ASSERT_ANY_THROW(
```

Comment:
Don't use `ANY_THROW`, but assert that you in fact get an exception with the expected error message ( `AD_EXPECT_THROW_WITH_MESSAGE` is your friend.